CMP1903_A1_2324/Testing.cs

```diff
@@ -1,20 +1,55 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
+using System.Runtime.InteropServices;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace CMP1903_A1_2324
 {
+    /// <summary>
+    /// Class <c>Testing()</c> provides methods to debug and check valid output for each die value and the sum.
+    /// </summary>
     internal class Testing
     {
-        /*
-         * This class should test the Game and the Die class.
-         * Create a Game object, call the methods and compare their output to expected output.
-         * Create a Die object and call its method.
-         * Use debug.assert() to make the comparisons and tests.
-         */
+        /// <summary>
+        /// Creating a reference to an object of the <c>Game</c> class.
+        /// </summary>
+        private Game game;
 
-        //Method
+        /// <summary>
+        /// Public method to get reference to the object.
+        /// </summary>
+        /// <returns>
+        /// The reference to the Game object.
+        /// </returns>
+        public Game GetGame() { return game; }
+
+        /// <summary>
+        /// A constructor to initialises a Game object for class <o>Testing</o>.
+        /// </summary>
+        public Testing()
+        {
+            game = new Game();
+        }
+
+        /// <summary>
+        /// Method <c>CheckRange()</c> to check if the value of each die is valid.
+        /// </summary>
+        public void CheckRange()
+        {
+            Debug.Assert(game.GetDie1().CurrentValue >= 1 && game.GetDie1().CurrentValue <= 6, "Invalid Die Value");
```

Comment:
Could these be split up for a more comprehensive testing experience?
If you were to do it this way it should allow you to find out what exactly is wrong with a roll, i.e you would instantly know if it's too high or too low?